pkg/exchange/coinbase/stream_handlers.go

```diff
@@ -276,7 +318,13 @@ func (s *Stream) handleMatchMessage(msg *MatchMessage) {
 		return
 	}
 	trade := msg.Trade()
-	s.EmitTradeUpdate(trade)
+	if _, fullExist := s.subLocalChannelsMap[fullChannel]; fullExist {
+		// the stream contains all matches, emit market trade
+		s.EmitMarketTrade(trade)
```

Comment:
public market trades are from the private user data stream?

Reply:
If the user subscribe to `full` channel, the feeds will contain all the matches on Coinbase Exchange.
In the other hand, if the user only subscribe to `user` without `full`, there will be only matches for the user.